demo/api/src/menus/menus.resolver.ts

```diff
@@ -50,10 +50,13 @@ export class MenusResolver {
         @Args("scope", { type: () => PageTreeNodeScope }) scope: PageTreeNodeScope,
         @RequestContext() { includeInvisiblePages }: RequestContextInterface,
     ): Promise<PageTreeNodeInterface[]> {
-        return this.pageTreeService
-            .createReadApi({
-                visibility: [PageTreeNodeVisibility.Published, ...(includeInvisiblePages || [])],
-            })
-            .pageTreeRootNodeList({ scope, category: PageTreeNodeCategory.TopMenu, excludeHiddenInMenu: true });
+        return (
+            this.pageTreeService
+                .createReadApi({
+                    visibility: [PageTreeNodeVisibility.Published, ...(includeInvisiblePages || [])],
+                })
+                //.preloadNodes(scope)
```

Comment:
```suggestion
                .preloadNodes(scope)
```

Was this intentional?

Reply:
no, we need to inject PageTreeReadApiService tough (instead of using createReadApi which would build a new cache)